apps/src/music/blockly/MusicBlocklyWorkspace.ts

```diff
@@ -383,10 +383,14 @@ export default class MusicBlocklyWorkspace {
    * Given the exact current playback position, get the start position of the trigger,
    * adjusted based on when the trigger should play (immediately, next beat, or next measure).
    */
-  getTriggerStartPosition(id: string, currentPosition: number) {
+  getTriggerStartPosition(
+    id: string,
+    currentPosition: number,
+    blockMode: ValueOf<typeof BlockMode>
```

Comment:
Not your change specifically, but wondering if we can clean this up a bit while we're here. `blockMode` is only used in this function to determine if we should return the currentPosition exactly, or rounded based on the `triggerStart` value we've stored in our `triggerIdToStartType` map. That map was only really created to handle the simple2 trigger block which has a dropdown for when the trigger should occur (immediately, next beat, next measure). For advanced mode, looks like we just store a value of `null` in this map because there's no 'trigger start' field for advanced trigger blocks. So it might be easier to simply store `TriggerStart.IMMEDIATELY` for trigger blocks that don't have a trigger start field (i.e. advanced trigger blocks) when we're compiling trigger events in `compileSong`. That way we don't need to branch based off of block mode here.

Reply:
Updated! https://github.com/code-dot-org/code-dot-org/pull/61127/commits/4017e5fdac91729575271a8af0fb88bb91ee1cd8